manifests/init.pp

```diff
@@ -72,8 +73,7 @@
   $manage_repos          = 1,
   $full_refresh_on_yum   = 0,
   $package_ensure        = 'latest',
-  $repo_extras           = false,
-  $repo_optional         = false
+  Optional[Array[String]] $enabled_repo_ids = [],
```

Comment:
Can you please use `Array[String[1]]` instead? The default for arrays should not be optional, but empty.  We documented that in [our review guidelines](https://voxpupuli.org/docs/#reviewing-a-module-pr)